src/main/java/com/nike/cerberus/service/AuthenticationService.java

```diff
@@ -247,6 +250,43 @@ private IamRoleAuthResponse authenticate(IamPrincipalCredentials credentials, Ma
                     .withExceptionMessage("Failed to write IAM role authentication response as JSON for encrypting.")
                     .build();
         }
+
+        // if the metadata and policies make the token too big to encrypt with KMS we can as a stop gap trim the metadata and policies from the token
+        // This information is stored in Vault and can be fetched by the client with a look-up self call
+        if (authResponseJson.length > KMS_SIZE_LIMIT) {
+            String originalMetadata = "unknown";
+            String originalPolicies = "unknown";
+            try {
+                originalMetadata = objectMapper.writeValueAsString(authResponse.getMetadata());
+                originalPolicies = objectMapper.writeValueAsString(authResponse.getPolicies());
+            } catch (JsonProcessingException e) {
+                logger.warn("Failed to serialize original metadata or policies for token generated for IAM Principal: {}", credentials.getIamPrincipalArn());
+            }
+
+            authResponse.setMetadata(ImmutableMap.of("_truncated", "true"));
+            authResponse.setPolicies(ImmutableSet.of("_truncated"));
+
+            logger.error(
+                    "The auth token has length: {} which is > {} KMS cannot encrypt it, truncating token and removing policies and metadata " +
```

Comment:
Is this statement right?  You aren't truncating the token, are you?  You are truncating the policies and metadata.